elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -174,57 +175,114 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
         }
     }
 
+    @SuppressWarnings("unchecked")
     private void populateInheritance(ElideTableConfig elideTableConfig) {
-        for (Table table : elideTableConfig.getTables()) {
+        //ensures validation is run before populate always.
+        validateInheritance(this.elideTableConfig);
+
+        List<Table> tables = elideTableConfig.getTables().stream().collect(Collectors.toList());
+        while (!tables.isEmpty()) {
+            Table table = tables.remove(0);
             if (table.hasParent()) {
-                Map<String, Measure> measures = getInheritedMeasures(table, new HashMap<String, Measure>());
+
+                Table parent = table.getParent(elideTableConfig);
+                // If parent also extends, ensure parent is processed first.
+                if (parent.getExtend() != null && parent.getSql() == null && parent.getTable() == null) {
+                    tables.add(table);
+                    continue;
+                }
+
+                Map<String, Measure> measures = (Map<String, Measure>) getInheritedAttribute(table,
+                        new HashMap<String, Measure>(), (tab, result) -> {
```

Comment:
You technically don't need to add a table parameter (tab) here because it is available through closure.

Reply:
Having both tab and table looks odd.